src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs

```diff
@@ -205,10 +205,378 @@ public void SearchPatternIgnoreSubDirectories()
             }
         }
 
+        [Theory,
+            // '[' should not matter, but is special to Unix matching APIs
+            InlineData(
+                "[foo]",
+                new string[] { @"f", @"o", @"o", @"foo", @"[foo]" },
+                new string[] { @"[foo]" }),
+            ]
+        public void PatternTests_UnixPatterns(string pattern, string[] sourceFiles, string[] expected)
+        {
+            string testDir = PrepareDirectory(sourceFiles);
+            ValidatePatternMatch(expected, GetEntries(testDir, pattern));
+        }
+
+        [ActiveIssue(20779, TestPlatforms.AnyUnix)]
+        [Theory,
+            // Question marks collapse (right) to periods
+            InlineData(
+                "f???.txt",
+                new string[] { @"f.txt", @"foo.txt", @"foob.txt", @"fooba.txt", @"foobar.txt" },
+                new string[] { @"f.txt", @"foo.txt", @"foob.txt" }),
+            // Question marks don't collapse to !periods
+            InlineData(
+                "???b??.txt",
+                new string[] { @"f.txt", @"foo.txt", @"foob.txt", @"fooba.txt", @"foobar.txt" },
+                new string[] { @"foob.txt", @"fooba.txt", @"foobar.txt" }),
+            // Question marks collapse (right) to end
+            InlineData(
+                "foo.t??",
+                new string[] { @"foo", @"foo.t", @"foo.tx", @"foo.txt", @"bar.txt", @"foo.txxt" },
+                new string[] { @"foo.t", @"foo.tx", @"foo.txt" }),
+            ]
```

Comment:
nit: this theory is huge. Can we break up the inlinedata into individual attributes like is done elsewhere?